src/java_bytecode/java_types.h

```diff
@@ -83,10 +81,8 @@ std::vector<typet> java_generic_type_from_string(
 typet java_bytecode_promotion(const typet &);
 exprt java_bytecode_promotion(const exprt &);
 
-bool is_java_array_tag(const irep_idt& tag);
-
-bool is_valid_java_array(const struct_typet &type);
-
+bool is_java_array_tag(const irep_idt &tag);
+bool is_valid_java_array(const struct_typet &);
```

Comment:
Just out of curiosity - is there a reason why the parameter name was removed from this (type) but not from the the line above (tag)?

Reply:
This one is borderline. The question is whether you can tell what the parameter is from the name of the function and the type of the parameter. The problem with irep_idt is that there are many uses of it; a struct_type is less ambiguous.